Continue cofactors true false implementation

hiandra opened the pull request.

hiandra committed:
[Print table function implemented]

   - Implemented a method from Manager to print on console
     the UniqueTable.

hiandra committed:
[Created test for CofatorTrue(f)]

   - Declaration of Cofactor functions.
   - Fixed nomenclature of the function addNodes() to
     be cammelCase like the others and removed it from
     ManagerInterface.h.
   - Implemented test for CofactorTrue(f): failed due
     to lack of implementation of the corresponding
     function.

hiandra committed:
[Implemented CofactorTrue(f)]

   - Created logic for the cofactorTrue function.
   - Altered its test to check more cases.
   - Test result: passed.

hiandra committed:
[Created test for CoFactorFalse(f)]

   - CoFactorFalse test result without implementation of
     the corresponding function: failed.

hiandra committed:
[Implementation of CoFactorFalse(f)]

   - After the implementation, the corresponding test
     result: passed.

hiandra committed:
[Created test for CoFactorTrue(f,x)]

   - Since there's no implementation yet for this funtion,
     test result: failed.

hiandra committed:
[Implemented CoFactorTrue(f,x)]

   - After the implementation, its test passed.
   - Reduced the functions coFactorTrue(f) and coFactorFalse(f)
     to run with less instructions since topVar of a variable is
     the same as its own ID.

hiandra committed:
[Created test for coFactorFalse(f,x)]

   - Since the function still wasn't implemented,
     the test failed.

hiandra committed:
[Implemented coFactorFalse(f,x)]

   - Optimized coFactor Functions to run with less instructions.
   - After implementation of its function: test passed.

MohamedShelkamy commented on src/Manager.cpp:
```diff
@@ -103,6 +101,49 @@ bool Manager::isConstant(BDD_ID f) {
+    if( unique_table[f].topvar != x )
+        return f;
+    else
+        return unique_table[topVar(f)].high;
```
i think this one needs to be unique_table[f].high

hiandra committed:
[Fixed bug in coFactors]

   - Fixed bug caught in pull request review.
   - Adjusted coFactors(f,x) to be able to solve cases
     where topVar(f) < x. Now they call ite, so it was
     declared, but not yet implemented.
   - Test cases adjusted and testing the cases where
     ite is called in coFactors is missing.

DavidFosca commented on src/Manager.cpp:
```diff
@@ -32,11 +32,11 @@ BDD_ID Manager::createVar(const std::string &label) {
-     BDD_node new_var(a,b,c,d, (std::string&)e);
-     unique_table.push_back(new_var);
-
+void Manager::addNode(BDD_ID a, BDD_ID b, BDD_ID c, BDD_ID d, std::string &e){
```
I think we could have put this inside the function that is calling it and for testing purposes used the SetUp() method from GTest.

MohamedShelkamy reviewed (approved):
Perfecto

hiandra committed:
[Removed verification of repeated vars]

   - When running all tests together, 2 coFactorTests were failing.
   - In previous commits they were run one by one and the problem
     wasn't noticed.
   - Removing the comparison between labels to check if the new variable
     is already created in createVar solved the problem.
   - While debugging, we noticed the var d was being removed from the
     table when adding the node a+b. But we didn't understand why
     removing the comparison solved it. We removed it because it wouldn't
     be useful for the rest of the implementation.
